pkg/apis/core/validation/validation.go

```diff
@@ -5005,6 +5005,16 @@ func ValidateSecretUpdate(newSecret, oldSecret *core.Secret) field.ErrorList {
 	}
 
 	allErrs = append(allErrs, ValidateImmutableField(newSecret.Type, oldSecret.Type, field.NewPath("type"))...)
+	if oldSecret.Immutable != nil && *oldSecret.Immutable {
+		if !reflect.DeepEqual(newSecret.Immutable, oldSecret.Immutable) {
+			allErrs = append(allErrs, field.Forbidden(field.NewPath("immutable"), "field is immutable when immutable is set"))
```

Comment:
It seems 'when immutable is set' is redundant

Reply:
I don't fully agree - if "immutable" is unset, this field is actually mutable.

Reply:
Agree with @wojtek-t 

field names in errors should be back-quoted: "when `immutable` is set"